### PR TITLE
Setup i18n for docusaurus

### PR DESCRIPTION
**Pull request by Kevan-Y**
## Issue This PR Addresses
fixes: #2922 

## Type of Change

- [ ] **Bugfix**: Change which fixes an issue
- [x] **New Feature**: Change which adds functionality
- [x] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description
Setup [i18n](https://docusaurus.io/docs/i18n/tutorial) for docusaurus, I only put languages that we are going to translate based on the table of this issue https://github.com/Seneca-CDOT/telescope/issues/2962

**Note: Only the overview page is overwritten, the rest is gonna stay in English**

## Steps to test the PR
Option 1: Run build
- `cd src/docs`
- `pnpm build && pnpm serve`

Option 2: Run dev
- `cd src/doc`
- `pnpm start -- --locale <LOCAL_CODE>` e.g. `pnpm start -- --locale fr`(For the french version)

https://user-images.githubusercontent.com/58233223/155567489-90c05c83-6e4a-4e4a-a54a-241d724a1900.mp4

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by Kevan-Y** [comment]
To add a new language later, we have to add the LOCAL_CODE in `docusaurus.config.js` in `i18n`

Then create your i18n folder.
`mkdir -p i18n/<LOCAL_CODE>/docusaurus-plugin-content-docs/current`
Then put your translated file in there. I believe it matches the structure of the docs file (where we contain all our original docs )

**Comment by aserputov** [comment]
Sounds good for me.

**Review by aserputov — APPROVED** [review]

**Review by tcvan0707 — APPROVED** [review]
Tested locally, it works for me. Later on, we would add the translation version in to the file inside `src/docs/i18n`, is that correct?

**Comment by Kevan-Y** [comment]
> Tested locally, it worked for me. Later on, we would add the translation version in to the file inside `src/docs/i18n`, is that correct?

yes, if you looked at the structure there is a folder for each languages we are translating for

**Review by DukeManh — APPROVED** [review]

**Review by cindyledev — APPROVED** [review]

File: src/docs/docusaurus.config.js [hunk 1]
```diff
@@ -34,6 +34,18 @@ const config = {
...
   ],
+  i18n: {
+    defaultLocale: 'en',
+    locales: ['en', 'fr', 'vi', 'ru', 'uk', 'es', 'zh-cn'],
+    localeConfigs: {
+      en: {
+        htmlLang: 'en-CA',
+      },
+      fr: {
+        htmlLang: 'fr-fr',
```
Comment (humphd) [inline src/docs/docusaurus.config.js:45]: `fr-CA`? 🇨🇦

**Review by humphd — APPROVED** [review]

File: src/docs/docusaurus.config.js (hunk 1 above)
Comment (Kevan-Y) [reply src/docs/docusaurus.config.js:45]: I don't know about the difference between fr-CA and fr-fr. I'm translating this page, so my French is from France

**Review by manekenpix — APPROVED** [review]

File: src/docs/docusaurus.config.js (hunk 1 above)
Comment (Kevan-Y) [reply src/docs/docusaurus.config.js:45]: I was following this, https://www.fincher.org/Utilities/CountryLanguageList.shtml.

**Review by cindyledev — APPROVED** [review]

**Pull request merged**